### PR TITLE
Different workshare calculations for rewarded vs active set

### LINKED ISSUE #2071: Integrate rewarded set into `tokenomics` rewarding

### PR DESCRIPTION
**Pull request by durch**
Closes nymtech/nym#2071 and nymtech/nym#2070

### CODE REVIEW

**Review by jstuczyn — COMMENTED** [review]
Conceptually looks good to me. However, I've got one question - how does this calculation update affect delegators rewarding? Should they also be adjusted?

**Comment by durch** [comment]
> Conceptually looks good to me. However, I've got one question - how does this calculation update affect delegators rewarding? Should they also be adjusted?

Overall the node reward will be lower, so everyone will get less, owner, and delegators

**Comment by jstuczyn** [comment]
> > Conceptually looks good to me. However, I've got one question - how does this calculation update affect delegators rewarding? Should they also be adjusted?
> 
> Overall the node reward will be lower, so everyone will get less, owner, and delegators

I might be misremembering, but wasn't it the case that delegators of nodes from the inactive set aren't getting anything?

**Comment by durch** [comment]
> > > Conceptually looks good to me. However, I've got one question - how does this calculation update affect delegators rewarding? Should they also be adjusted?
> > 
> > 
> > Overall the node reward will be lower, so everyone will get less, owner, and delegators
> 
> I might be misremembering, but wasn't it the case that delegators of nodes from the inactive set aren't getting anything?

I'm not sure, will tap Claudia

**Comment by durch** [comment]
> > > > Conceptually looks good to me. However, I've got one question - how does this calculation update affect delegators rewarding? Should they also be adjusted?
> > > 
> > > 
> > > Overall the node reward will be lower, so everyone will get less, owner, and delegators
> > 
> > 
> > I might be misremembering, but wasn't it the case that delegators of nodes from the inactive set aren't getting anything?
> 
> I'm not sure, will tap Claudia

Good thing I did, it slipped my mind completely that this has changed, and we did not capture the change with the ticket. Double checked with Claudia, and we should be good

**Comment by jstuczyn** [comment]
One more thing I forgot to mention - we probably also need to update validator API and how it calculates all those values as currently it only sends rewards to the **CURRENT** rewarded set

**Comment by durch** [comment]
> One more thing I forgot to mention - we probably also need to update validator API and how it calculates all those values as currently it only sends rewards to the **CURRENT** rewarded set

I know what you mean, we need to ticket that up, but we also need to discuss how we'll go about actually updating the rewarded set at required intervals

File: validator-api/src/rewarding/mod.rs [hunk 1]
```diff
@@ -158,21 +185,18 @@ impl Rewarder {
...
-        info!("-- Reward pool: {} unym", reward_pool);
-        info!("---- Epoch reward pool: {} unym", period_reward_pool);
-        info!("-- Circulating supply: {} unym", circulating_supply);
+        info!("-- Reward pool: {} unym", epoch_reward_params.reward_pool);
+        info!(
+            "---- Epoch reward pool: {} unym",
+            epoch_reward_params.period_reward_pool
+        );
+        info!(
+            "-- Circulating supply: {} unym",
```
Comment (neacsu) [inline validator-api/src/rewarding/mod.rs]: I think it would be better to use `DENOM` for those as well.

**Pull request merged**